pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Markers.java

```diff
@@ -290,6 +306,14 @@ private static ByteBuf newTxnMarker(MarkerType markerType, long sequenceId, long
         MessageMetadata msgMetadata = msgMetadataBuilder.build();
 
         ByteBuf payload = PooledByteBufAllocator.DEFAULT.buffer();
+        if (messageIdData.isPresent()) {
+            PulsarMarkers.TxnCommitMarker commitMarker =
+                PulsarMarkers.TxnCommitMarker.newBuilder().setMessageId(messageIdData.get()).build();
+            int size = commitMarker.getSerializedSize();
+            payload = PooledByteBufAllocator.DEFAULT.buffer(size);
```

Comment:
you are allocating two payload here. and the previous payload is not even released.

Reply:
fixed.